derive_client/data_types/models.py

```diff
@@ -130,25 +183,21 @@ def status(self) -> TxStatus:
         return TxStatus.ERROR
 
 
-@dataclass
+@dataclass(config=ConfigDict(validate_assignment=True))
 class BridgeTxResult:
+    currency: Currency
     bridge: BridgeType
     source_chain: ChainID
     target_chain: ChainID
     source_tx: TxResult
-    target_tx: TxResult
     target_from_block: int
+    target_tx: TxResult | None = None
 
     @property
     def status(self) -> TxStatus:
-        statuses = [self.source_tx.status, self.target_tx.status]
-        if all(s == TxStatus.SUCCESS for s in statuses):
-            return TxStatus.SUCCESS
-        if any(s == TxStatus.FAILED for s in statuses):
-            return TxStatus.FAILED
-        if any(s == TxStatus.PENDING for s in statuses):
-            return TxStatus.PENDING
-        return TxStatus.ERROR
+        if self.source_tx.status is not TxStatus.SUCCESS:
+            return self.source_tx.status
+        return self.target_tx.status if self.target_tx is not None else TxStatus.PENDING
```

Comment:
initially we initialized `BridgeTxResult.target_tx` as "empty" (`TxResult("", None, None)`), but `TxResult` should not exist before a valid `tx_hash` is retrieved. Since `target_tx` can now be `None`, and since this was erroneous (since would yield ERROR when initialzing `target_tx` as "empty") this is not correct. As a bonus, the logic here is much cleaner and clearer now.